sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -1054,5 +1056,329 @@ public virtual Response<SearchSuggestion[]> GetSearchSuggestions(string searchSt
                 throw;
             }
         }
+
+        /// <summary>
+        /// Gets time series insight types in pages asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{TimeSeriesType}"/> of Time Series instances types with the http response.</returns>
+        public virtual AsyncPageable<TimeSeriesType> GetTimeSeriesTypesAsync(
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetTimeSeriesTypes)}");
+            scope.Start();
+
+            try
+            {
+                async Task<Page<TimeSeriesType>> FirstPageFunc(int? pageSizeHint)
```

Comment:
I don't see the pageSizeHint being used anywhere. Does the service not support it?

Reply:
Nor do I see it used in the nextPageFunc

Reply:
That's a very good observation, I'll follow up with Tsi team and update this code accordingly.